src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -407,16 +412,25 @@ def _sync_public_model_to_hub(self, model: JumpStartModelInfo, thread_num: int):
             tags=tags,
         )
 
-    def _fetch_studio_specs(self, model_specs: JumpStartModelSpecs) -> Dict[str, Any]:
-        """Fetches StudioSpecs given a model's SDK Specs."""
-        model_id = model_specs.model_id
-        model_version = model_specs.version
+    def _fetch_studio_specs(self, studio_spec_path: str) -> Dict[str, Any]:
+        """Fetches StudioSpec given spec path."""
 
-        key = utils.generate_studio_spec_file_prefix(model_id, model_version)
+        response = self._s3_client.get_object(
+            Bucket=utils.get_jumpstart_content_bucket(self.region), Key=studio_spec_path
+        )
+        return json.loads(response["Body"].read().decode("utf-8"))
+    
+    def _fetch_manifest_from_s3(self, key: str) -> List[Dict[str,Any]]:
         response = self._s3_client.get_object(
             Bucket=utils.get_jumpstart_content_bucket(self.region), Key=key
         )
         return json.loads(response["Body"].read().decode("utf-8"))
+    
+    def _find_entry_in_studio_manifest(self, studio_manifest: List[Dict[str,Any]], model_id: str) -> Dict[str,Any]:
```

Comment:
can we create a mapping b/w model id and manifest entry? so we don't have to iterate each time?

Reply:
Good point, will update.